middleware/auth.js

```diff
@@ -0,0 +1,32 @@
+import jwt from "jsonwebtoken"
+import asyncHandler from "./async";
+import ErrorResponse from "../utils/errorResponse";
+import User from "../model/User";
+
+
+// Protect routes
+exports.protect = asyncHandler(async (req, res, next) => {
+    const authHeader = req.headers.authorization;
+    if (!authHeader || !authHeader.startsWith("Bearer ")){
+        return next(new ErrorResponse("No token provided", 401))
+    }
+
+    const token = authHeader.split(" ")[1];
+
+    // Make sure token exists
+    if(!token) {
+        return next(new ErrorResponse('No token provided', 401));}
```

Comment:
it would be more clean and readable if we had a file for constants and add all const strings to it